### DISCUSSION TITLE
Difficulty Controlling Regions at High Zoom Levels

### DISCUSSION THREAD

**Discussion by jake-mcauley** [katspaugh/wavesurfer.js#3572] [Q&A]
I seem to have an issue with WaveSurfer.js when working with regions at high zoom levels. Specifically, the issue arises when the zoom level increases to a point where the region's width is less than the visible area of the waveform.

When zoomed in, I would like to be able to click on a region and adjust its position on the waveform from the clicked location, without affecting the waveform's scroll position.

Currently, upon clicking the region under these conditions, the waveform automatically scrolls, centring the region in the view. This behaviour prevents precise adjustments of the region's boundaries, as it shifts the view unexpectedly and does not allow for direct manipulation of the region from the clicked point.

Steps to Reproduce:
1. Create a region
2. Zoom into the waveform until the width of a region is less than the width of the waveform container.
3. Scroll to the left to view the start of the region.
4. Click on the region to drag its position.
5. Observe that the waveform scrolls to center the region, rather than allowing for adjustment from the click point.

**Comment by katspaugh** [comment]
The default click behavior for regions is to pass the click through to the waveform. You can prevent it by stopping the event propagation like so:

```js
  regionsInstance.on('region-clicked', (region, e) => {
    e.stopPropagation() // prevent triggering a click on the waveform
    region.play() // Do stuff
  })
```

**Reply by jake-mcauley** [reply]
Sorry, I may not have explained this very well. The issue is when the region is selected as opposed to being clicked. Here is a video to help explain it.


https://github.com/katspaugh/wavesurfer.js/assets/60385455/43ce4b09-2fe1-411e-97dc-01b573e5f567

**Reply by katspaugh** [reply]
I see. Try adding autoCenter: false or autoScroll: false (or both). See https://wavesurfer.xyz/docs/types/wavesurfer.WaveSurferOptions
